src/renderer/components/SplittedChatListAndView.js

```diff
@@ -20,20 +20,20 @@ class SplittedChatListAndView extends React.Component {
   constructor (props) {
     super(props)
     this.state = {
-      chatId: null
+      selectedChat: {}
     }
 
     this.onChatClick = this.onChatClick.bind(this)
   }
 
   onChatClick (chat) {
```

Comment:
I think the main problem starts here. If `onChatClick()` called back with a `chatId` instead all caching problems would be gone :)

Reply:
Nevermind my comments for now. I'll PR it later once this is merged.